### PR TITLE
Typespecs

### PR DESCRIPTION
**Pull request by anthonator**
This is a WIP but I wanted to get this up to make sure what I have for `Mail.Message` looks good.

### CODE REVIEW

File: lib/mail/message.ex [hunk 1]
```diff
@@ -4,6 +4,14 @@ defmodule Mail.Message do
             parts: [],
             multipart: false
 
+  @type t ::
+          %__MODULE__{
```
Comment (bcardarella) [inline lib/mail/message.ex]: can you break these out into their own types? The types themselves can be similar to here, just that these types will be necessary for the functional specs. Similar to what I did here: https://github.com/DockYard/openid_connect/blob/master/lib/openid_connect.ex

Comment (anthonator) [reply lib/mail/message.ex]: Thanks for the feedback. I broke out the struct fields into individual types.

**Comment by anthonator** [comment]
@bcardarella I added typespecs to the `Mail` module. Let me know what you think.

**Comment by princemaple** [comment]
Also, still waiting for a release 😉

**Comment by bcardarella** [comment]
in anticipation of the hopefully soon gradual typing system in Elixir I'll close this, sorry I never got around to merging it.

**Pull request closed without merging**